Generate IPC serialization for WebCore::UserStyleLevel enumeration

zdobersek opened the pull request:
#### ead871fa35c6ac44d0eda9c5a5660eb62bb7e2d5
<pre>
Generate IPC serialization for WebCore::UserStyleLevel enumeration
<a href="https://bugs.webkit.org/show_bug.cgi?id=264403">https://bugs.webkit.org/show_bug.cgi?id=264403</a>

Reviewed by Chris Dumez and Michael Catanzaro.

Turn the WebCore::UserStyleLevel enumeration into a scoped one, adjusting uses
of values accordingly. This enables removing the EnumTraits specialization in
favor of better-handled IPC serialization specification.

* Source/WebCore/dom/ExtensionStyleSheets.cpp:
(WebCore::createExtensionsStyleSheet):
(WebCore::ExtensionStyleSheets::pageUserSheet):
* Source/WebCore/page/UserStyleSheet.h:
(WebCore::UserStyleSheet::UserStyleSheet):
* Source/WebCore/page/UserStyleSheetTypes.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/Cocoa/_WKUserStyleSheet.mm:
(-[_WKUserStyleSheet initWithSource:forMainFrameOnly:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKUserStyleSheetInternal.h:
(API::toWebCoreUserStyleLevel):
* Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
(toUserStyleLevel):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::addInjectedContent):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionDynamicScriptsCocoa.mm:
(WebKit::WebExtensionDynamicScripts::injectStyleSheets):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::addUserStyleSheet):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(+[WebView _addUserStyleSheetToGroup:world:source:url:includeMatchPatternStrings:excludeMatchPatternStrings:injectedFrames:]):

Canonical link: <a href="https://commits.webkit.org/270452@main">https://commits.webkit.org/270452@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33c1b292e8f8bdfeb3f8c72c710d143dbd8d99bf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25553 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4158 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26838 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27657 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23419 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/25835 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5898 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1593 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23567 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25802 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3083 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22030 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28239 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/2724 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22981 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/29076 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23319 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23346 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/26912 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2726 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/980 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4108 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6123 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3184 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3063 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/UserStyleSheetTypes.h:
```diff
@@ -25,14 +25,16 @@
 
 enum UserStyleInjectionTime { InjectInExistingDocuments, InjectInSubsequentDocuments };
-enum UserStyleLevel { UserStyleUserLevel, UserStyleAuthorLevel };
+enum class UserStyleLevel : uint8_t {
```
`: bool`

cdumez commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -6568,3 +6568,9 @@ struct WebCore::KeypressCommand {
 #endif
+
+header: <WebCore/UserStyleSheetTypes.h>
+[Nested] enum class WebCore::UserStyleLevel : uint8_t {
```
`: bool`

cdumez commented on Source/WebCore/page/UserStyleSheetTypes.h:
```diff
@@ -25,14 +25,16 @@
 
 enum UserStyleInjectionTime { InjectInExistingDocuments, InjectInSubsequentDocuments };
-enum UserStyleLevel { UserStyleUserLevel, UserStyleAuthorLevel };
+enum class UserStyleLevel : bool {
```
nit: could have stayed on a single line, it is our preferred style for small enums.

zdobersek replied:
OK.

zdobersek committed:
Generate IPC serialization for WebCore::UserStyleLevel enumeration
https://bugs.webkit.org/show_bug.cgi?id=264403

Reviewed by Chris Dumez and Michael Catanzaro.

Turn the WebCore::UserStyleLevel enumeration into a scoped one, adjusting uses
of values accordingly. This enables removing the EnumTraits specialization in
favor of better-handled IPC serialization specification.

* Source/WebCore/dom/ExtensionStyleSheets.cpp:
(WebCore::createExtensionsStyleSheet):
(WebCore::ExtensionStyleSheets::pageUserSheet):
* Source/WebCore/page/UserStyleSheet.h:
(WebCore::UserStyleSheet::UserStyleSheet):
* Source/WebCore/page/UserStyleSheetTypes.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/Cocoa/_WKUserStyleSheet.mm:
(-[_WKUserStyleSheet initWithSource:forMainFrameOnly:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKUserStyleSheetInternal.h:
(API::toWebCoreUserStyleLevel):
* Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
(toUserStyleLevel):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::addInjectedContent):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionDynamicScriptsCocoa.mm:
(WebKit::WebExtensionDynamicScripts::injectStyleSheets):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::addUserStyleSheet):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(+[WebView _addUserStyleSheetToGroup:world:source:url:includeMatchPatternStrings:excludeMatchPatternStrings:injectedFrames:]):

Canonical link: https://commits.webkit.org/270452@main
